### PR TITLE
feat: improve module formats (cjs, esm, iife)

### PR DESCRIPTION
**Pull request by kettanaito**
- Adopts `tsup` over `rollup`.
- Distributes `/node` in CJS and ESM.
- Distributes `/native` in CJS and ESM.

### CODE REVIEW

File: config/scripts/postinstall.js [hunk 1]
```diff
@@ -13,18 +13,19 @@ const packageJson = JSON.parse(
...
 const { workerDirectory } = packageJson.msw
 const absoluteWorkerDirectory = path.resolve(parentPackageCwd, workerDirectory)
 
 if (!fs.existsSync(absoluteWorkerDirectory)) {
-  return console.error(
+  console.error(
     `[MSW] Failed to automatically update the worker script at "%s": given path does not exist.`,
     workerDirectory,
   )
+  process.exit()
```
Comment (kettanaito) [inline config/scripts/postinstall.js]: I think this is a bad idea. These cases are meant to halt early but don't exit the process (`npm install`).

**Comment by kettanaito** [comment]
## Released: v0.41.0 🎉

This has been released in v0.41.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v0.41.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/0.41.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by nvh95** [comment]
@kettanaito Can you briefly ELI5 the benefit of replacing `rollup` with `tsup`? Thank you.

**Comment by ryota-murakami** [comment]
@kettanaito I have a abandoned package that has never new release over 2 years.
This configuration will be very helpful for me to make a modern formant release! 😄
@nvh95 your jest-preview too! 🚀

**Comment by kettanaito** [comment]
Hey, @nvh95. Absolutely. 

The main reason to adopt `tsup` (which, transiently, adopts `esbiuld`) is the build speed. By that bundler migration alone I've reduced the build speed on CI from ~1m 50s to 7s. I'm lazy to do the math but that looks like a ginormous improvement to me. 

Other than that I find Rollup and esbuild almost synonymous in terms of consumer experience. I can write plugins for both, they certainly get the job done. 

@ryota-murakami, I'd be glad if it helps you in any way. Make those releases happen!

**Comment by ryota-murakami** [comment]
@kettanaito Thank you! 😄

**Comment by nvh95** [comment]
Thank you for the explanarion.
